main.go

```diff
@@ -29,6 +32,36 @@ func main() {
 		os.Exit(1)
 	}
 
+	results, err := run(params)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Test Service Error: %v\n", err)
+		os.Exit(1)
+	}
+
+	if !results.OK() {
+		os.Exit(1)
+	}
+}
+
+func run(params commandParams) (*ldtest.Results, error) {
+	if params.suppressFailures != "" {
+		file, err := os.Open(params.suppressFailures)
+		if err != nil {
+			return nil, fmt.Errorf("cannot open provided suppression file: %v", err)
+		}
```

Comment:
Not super important because the program will be terminating anyway, but I think you want `defer file.Close()` here rather than putting `Close` at the end, since you would want it to be closed if there was an early return due to a scanner error.